qiskit/extensions/standard/swap.py

```diff
@@ -54,11 +54,16 @@ def reapply(self, circ):
 
 def swap(self, ctl, tgt):
     """Apply SWAP from ctl to tgt."""
-    if isinstance(ctl, QuantumRegister) and \
-            isinstance(tgt, QuantumRegister) and len(ctl) == len(tgt):
+    if isinstance(ctl, QuantumRegister):
+        ctl = [(ctl, i) for i in range(len(ctl))]
+    if isinstance(tgt, QuantumRegister):
+        tgt = [(tgt, i) for i in range(len(tgt))]
+
+    if ctl and tgt and isinstance(ctl, list) and \
+       isinstance(tgt, list) and len(ctl) == len(tgt):
```

Comment:
This should throw an error if `len(ctrl) != len(tgt)` rather than silently ignore.